packages/frontend/core/src/desktop/pages/workspace/index.tsx

```diff
@@ -110,6 +117,17 @@ export const Component = (): ReactElement => {
   }, [listLoading, meta, workspaceNotFound, workspacesService]);
 
   if (workspaceNotFound) {
+    if (BUILD_CONFIG.isElectron && searchParams.get('server')) {
+      const redirectUrl = window.location.href.replace(
+        window.location.origin,
+        ''
+      );
+      return (
+        <AffineOtherPageLayout>
+          <SignIn redirectUrl={redirectUrl} />
+        </AffineOtherPageLayout>
+      );
+    }
     if (detailDocRoute) {
       return (
         <SharePage
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: line +141]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Should consider selfhosted?<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9231?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
Server will be stripped off in the second attempt and reach to this point.